### PR TITLE
Bug 1119731 - gaia-build (Gb) and gaia-build-unit tests (Gbu) are marked as successful even though they contain test failures

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1119731

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e043e5bd4766)

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ suite('Keyboard layouts building tests', function() {
   // Build with all layouts with no dictionaries
   // XXX: bug 1125703: disable this test until bug 1119731's
   //      false-negative is resolved.
-  test.skip('APP=keyboard GAIA_KEYBOARD_LAYOUTS=noPreloadDictRequired ' +
+  test('APP=keyboard GAIA_KEYBOARD_LAYOUTS=noPreloadDictRequired ' +
```
Comment (julienw) [inline apps/keyboard/test/build/integration/keyboard_test.js]: are these changes expected ? :)

Comment (julienw) [inline apps/keyboard/test/build/integration/keyboard_test.js]: ah yeah, sorry, got it :)

Comment (julienw) [inline apps/keyboard/test/build/integration/keyboard_test.js]: I would have changed a test in build/test/ instead, that's why I was confused

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1503f0698513)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=06611d93755a)

File: build/utils-node.js [hunk 2]
```diff
@@ -78,8 +78,10 @@ function Commander(cmd) {
     //      pattern inside each script or find a sync module which doesn't
     //      require recompile again since TPBL doesn't support that.
     sh(cmds, function(err, stdout, stderr) {
-      if (err === null && typeof callback === 'function') {
-        callback(stdout);
+      if (err === null) {
+        callback && callback(0);
+      } else {
+        callback && callback(-1);
```
Comment (RickyChien) [inline build/utils-node.js]: http://www.tldp.org/LDP/abs/html/exitcodes.html
I prefer to use exit code 1 as "general errors"

File: build/utils-xpc.js [hunk 3]
```diff
@@ -1003,11 +1003,12 @@ function Commander(cmd) {
       log('cmd', command + ' ' + args.join(' '));
       process.init(_file);
       process.run(true, args, args.length);
+      callback && callback(process.exitValue);
     } catch (e) {
+      callback && callback(-1);
```
Comment (RickyChien) [inline build/utils-xpc.js]: same

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=abc58e4edb26)

**Pull request merged**